runtime/http_client.go

```diff
@@ -21,32 +21,68 @@
 package zanzibar
 
 import "net/http"
-import "go.uber.org/zap"
+import (
+	"net"
+	"time"
+
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+)
 
 // HTTPClient defines a http client.
 type HTTPClient struct {
 	gateway *Gateway
 
 	Client  *http.Client
 	Logger  *zap.Logger
+	Scope   *tally.Scope
 	BaseURL string
 }
 
 // NewHTTPClient will allocate a http client.
 func NewHTTPClient(
 	gateway *Gateway, baseURL string,
```

Comment:
we can add an extra argument here for httpClientName.

Then update the template ( https://github.com/uber/zanzibar/blob/master/codegen/templates/http_client.tmpl#L37 ), should be `"{{.Name}}"`